### ISSUE TITLE
Metadata not found in metadata location for table with Flink, Nessie and Trino

### ISSUE THREAD

**Issue by sheveg** [trinodb/trino#21007]
When using Flink to put data in iceberg using Nessie as catalog, I get an `Metadata not found in metadata location for table`  error when trying to query the data generated by flink. When I query the data generated by trino itself using its CLI, then I do not have this problem. The iceberg connector configurations is:
```yaml
iceberg: |-
      connector.name=iceberg
      iceberg.catalog.type=nessie
      iceberg.nessie-catalog.uri=http://nessie.nessie.svc.cluster.local:19120/api/v1
      iceberg.nessie-catalog.default-warehouse-dir=s3://steadops-playground-bucket
      iceberg.nessie-catalog.ref=main
      iceberg.register-table-procedure.enabled=true
      iceberg.file-format=PARQUET
      hive.s3.path-style-access=true
      hive.s3.endpoint=http://minio.minio.svc.cluster.local:80
      hive.s3.aws-access-key=${ENV:MINIO_ACCESS_KEY}
      hive.s3.aws-secret-key=${ENV:MINIO_SECRET_KEY}
      hive.s3.ssl.enabled=false
```

When I create the data using the trino CLI, the `.parquet` file is named in the schema: `<date>_<time>_<uuid>.parquet` whereas the data files generated by flink are named as `00000_0_<uuid>.parquet`. The metadata files are named according to the same name schema as far as I see: <seq_number>_<uuid>.metadata.json. The only difference I see, that the metadata directory for flink does not contain any `*.stats` files.

**Comment by ajantha-bhat** [comment]
I don't think it is related or specific to Nessie catalog. Can you try with other catalogs? 

I would also double check the catalog configurations at Trino side whether it is same as what is configured at the Flink side.

**Comment by rehevkor5** [comment]
I am seeing this issue with an Iceberg table that uses Hive Metastore. The `metadata_location` and `previous_metadata_location` point to files that don't exist.

But, what makes me suspicious is:

- the current-snapshot-timestamp-ms value is 4 days ago - but we are making ongoing commits to this table, more than once an hour
- the value for those fields has not changed since I started looking at them like 30 minutes ago

Is Trino doing some kind of caching of this data? If so, it is too aggressive or needs to be improved. The cache would need to use table metadata from within the past few hours (at the oldest) in order work reasonably well... And the older the cached data is, the staler the results would be (it would be querying from an old snapshot).

**Comment by ajantha-bhat** [comment]
@rehevkor5: Thanks for confirming that this is `not` a nessie catalog issue. 

cc: @ebyhr, @findepi

**Comment by Souldiv** [comment]
hi @rehevkor5 how were you able to mitigate this issue? I am facing the same problem in my cluster

**Comment by rehevkor5** [comment]
@Souldiv I had to get some help from a coworker. His response was that "MySQL metastore RDS has the correct metadata, but I suspect it is an issue with glue metadata sync on datalake side". Essentially, our Trino is configured to use an AWS Glue replica of the Hive Metastore contents, for various reasons. The Glue sync was failing due to an error on the API call to AWS:

> Unable to alter table. Number of TABLE_VERSION resources exceeds the account limit xxx

That resulted in Trino seeing outdated table metadata. The main fix was to start setting "SkipArchive=true" when making the `UpdateTable` call to Glue.

@ajantha-bhat I don't think you can assert that based on my example. There could be more than one way to encounter this issue.
